### PR TITLE
Update nin readme, various setup fixes.

### LINKED ISSUE #374: `nin run` crashes in new projects
If you have just created a new project using `nin new`, and made an initial commit, `nin run` will fail with the following error:

```
$ nin run
Found nin.json in project root (/Users/stiaje/Projects/test-nin)
SockJS v0.3.16 bound to "/socket"
child_process.js:524
    throw err;
    ^

Error: Command failed: git config --get remote.origin.url

    at checkExecSyncError (child_process.js:481:13)
    at execSync (child_process.js:521:13)
    at Object.getProjectMetadata (/Users/stiaje/Projects/nin/nin/backend/utils.js:12:20)
    at /Users/stiaje/Projects/nin/nin/backend/serve.js:145:22
    at /Users/stiaje/Projects/nin/nin/node_modules/graceful-fs/graceful-fs.js:43:10
    at /Users/stiaje/Projects/nin/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqWrap.oncomplete (fs.js:114:15)
```

The work-around for this error is setting a remote for your local git repo, but this should not be a requirement.

### PR DESCRIPTION
**Pull request by aleksanb**
Fixes #374 among other things.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+nin is Ninjadev's internal demo tool.
+It is a tool for easing development of browser-based WebGL demos.
+Core features:
 
-This project has a node backend that keeps track of all files and compiles files as they are edited.
-The frontend of this project is written in Angular and displays among other the layers that the demo consists of.
+- Node-based demotool, effortlessly reuse your effects, scenes, and create crazy transitions.
+- Livereloading of shaders and scenes in the browser. No more manual recompilation!
+- Tight THREE.js integration for all your WebGL needs.
+- Compile and pack your WebGL demo to a html.png file for easy compo delivery
```
Comment (stianjensen) [inline README.md]: .png.html

File: README.md [hunk 2]
```diff
@@ -21,20 +28,35 @@ This projects requires node version `7.1.0` or newer.
...
-
-Nin just became node-based instead of layer-based.
-Here is a list of gotchas to watch out for if you are used to layer-based nin:
+## Shaders
 
-- Changing res/graph.json does *not* trigger updates in a running nin instance.
+Create a shader by clicking `Generate -> Shader Node` in the frontend menu.
+It will be placed in the folder `src/shaders/nameOfTheShader`.
+To get livereload on shader change, you shader must be specified in the options object of your node in `res/graph.json`,
+the shader generator will do this for you.
```
Comment (stianjensen) [inline README.md]: Actually it won't, but that's a bug so it is fine that we say so here. I can fix the bug.

**Comment by cristeahub** [comment]
I read through the readme as a whole and I saw that in the developing section we don't mention the need to make an alias to run with your local version of nin. This will mean that people who install nin globally won't use their local version of nin. Could you add a section about that?

**Comment by stianjensen** [comment]
> I read through the readme as a whole and I saw that in the developing section we don't mention the need to make an alias to run with your local version of nin. This will mean that people who install nin globally won't use their local version of nin. Could you add a section about that?

You should use `npm link` for this. That should also properly switch from npmjs-installed to locally installed I believe.

**Comment by aleksanb** [comment]
Updated PR, removed my node generation fix in favor of @stianjensen s

**Comment by stianjensen** [comment]
Can you change the title to reflect all the changes in here?

**Comment by aleksanb** [comment]
Updated @stianjensen

File: nin/backend/utils.js [hunk 3]
```diff
@@ -6,10 +6,25 @@ const execSync = require('child_process').execSync;
...
 
 function getProjectMetadata(projectPath) {
+  let projectSettings = ps.load(projectPath);
+  let projectOrigin;
+  let projectVersion;
+  try {
+    projectOrigin = execSync('git config --get remote.origin.url', {
+      stdio: ['ignore', 'pipe', 'ignore']
+    });
+    projectVersion = execSync('got rev-parse HEAD', {
```
Comment (stianjensen) [inline nin/backend/utils.js]: s/got/git/

**Pull request merged**